community/kernel/src/main/java/org/neo4j/kernel/impl/api/ConstraintEnforcingEntityOperations.java

```diff
@@ -378,24 +373,23 @@ public long nodeGetFromUniqueIndexSeek(
 
         // If we find the node - hold a shared lock. If we don't find a node - hold an exclusive lock.
         // If locks are deferred than both shared and exclusive locks will be taken only at commit time.
-        Locks.Client locks = state.locks().optimistic();
-        LockTracer lockTracer = state.lockTracer();
+        StatementLocks statementLocks = state.locks();
         long indexEntryId = indexEntryResourceId( labelId, predicates );
 
-        locks.acquireShared( lockTracer, INDEX_ENTRY, indexEntryId );
+        statementLocks.uniquenessConstraintEntryAcquireShared( indexEntryId );
```

Comment:
If we are taking a stab at a fundamental change in our locking model, maybe it's time to introduce `Update` locks as well, instead of doing little dance of grabbing and releasing locks?